### ISSUE TITLE
OpenGL Interop in OS X

### ISSUE THREAD

**Issue by kwonoh** [kylelutz/compute#328]
Several compile time errors occur in function [opengl_create_shared_context()](https://github.com/kylelutz/compute/blob/master/include/boost/compute/interop/opengl/context.hpp#L37)  on OS X (10.10.1, Xcode 6.1.1) when I try to compile [opengl_sphere example](https://github.com/kylelutz/compute/blob/master/example/opengl_sphere.cpp).

the errors looks like below:
```
../../../local/include/boost/compute/interop/opengl/context.hpp:59:35: error: C++ requires a type specifier for all declarations
    const cl_context_properties*, cl_gl_context_info, size_t, void*, size_t*);
                                  ^~~~~~~~~~~~~~~~~~
../../../local/include/boost/compute/interop/opengl/context.hpp:76:5: error: unknown type name 'CGLContextObj'
    CGLContextObj cgl_current_context = CGLGetCurrentContext();
    ^
../../../local/include/boost/compute/interop/opengl/context.hpp:76:41: error: use of undeclared identifier 'CGLGetCurrentContext'
    CGLContextObj cgl_current_context = CGLGetCurrentContext();
                                        ^
../../../local/include/boost/compute/interop/opengl/context.hpp:77:5: error: unknown type name 'CGLShareGroupObj'
    CGLShareGroupObj cgl_share_group = CGLGetShareGroup(cgl_current_context);
    ^
../../../local/include/boost/compute/interop/opengl/context.hpp:90:7: error: use of undeclared identifier 'CL_CONTEXT_PROPERTY_USE_CGL_SHAREGROUP_APPLE'
      CL_CONTEXT_PROPERTY_USE_CGL_SHAREGROUP_APPLE,
      ^
../../../local/include/boost/compute/interop/opengl/context.hpp:104:39: error: use of undeclared identifier 'CL_CURRENT_DEVICE_FOR_GL_CONTEXT_KHR'
      GetGLContextInfoKHR(properties, CL_CURRENT_DEVICE_FOR_GL_CONTEXT_KHR,
```
I handle the errors as follows:
1. [Add required headers](https://github.com/kylelutz/compute/pull/327)
2. I used below code to create shared context.
But it need to be done with more appropriate way. (e.g., checking GL context has been created, checking CL-GL sharing capable device)
```cpp
inline context opengl_create_shared_context()
#if defined(__APPLE__)
{
  CGLContextObj kCGLContext = CGLGetCurrentContext();
  CGLShareGroupObj kCGLShareGroup = CGLGetShareGroup(kCGLContext);

  cl_context_properties properties[] = {
    CL_CONTEXT_PROPERTY_USE_CGL_SHAREGROUP_APPLE,
    (cl_context_properties)kCGLShareGroup, 0};

  return context(clCreateContext(properties, 0, 0, NULL, 0, 0));
}
#else
// existing codes ...
```

**Comment by kylelutz** [comment]
Alright, this should be fixed in PR #425. Can you guys try that out?

**Comment by kylelutz** [comment]
This should be resolved as of d4adfc5feb4eb38eaf2786a7de00b74065cfa753. Please re-open if you still have issues.
